internal/wasm/memory.go

```diff
@@ -222,6 +232,20 @@ func (m *MemoryInstance) Grow(delta uint32) (result uint32, ok bool) {
 	newPages := currentPages + delta
 	if newPages > m.Max || int32(delta) < 0 {
 		return 0, false
+	} else if m.allocator != nil {
+		buffer := m.allocator.Grow(MemoryPagesToBytesNum(newPages))
+		if m.Shared {
+			if unsafe.SliceData(buffer) != unsafe.SliceData(m.Buffer) {
+				panic("shared memory cannot move")
```

Comment:
Would it make sense to `return 0, false` here?

Reply:
No, at this point the memory has moved, the harm is done.
I'll document instead that if you want to use your allocator for shared memories, it can't move the memory.